pkg/controller/clustersync/clustersync_controller.go

```diff
@@ -162,20 +161,34 @@ func NewReconciler(mgr manager.Manager, rateLimiter flowcontrol.RateLimiter) (*R
 	log.WithField("reapplyInterval", reapplyInterval).Info("Reapply interval set")
 	c := controllerutils.NewClientWithMetricsOrDie(mgr, ControllerName, &rateLimiter)
 	return &ReconcileClusterSync{
-		Client:                c,
-		logger:                logger,
-		reapplyInterval:       reapplyInterval,
-		resourceHelperBuilder: resourceHelperBuilderFunc,
+		Client:          c,
+		logger:          logger,
+		reapplyInterval: reapplyInterval,
 		remoteClusterAPIClientBuilder: func(cd *hivev1.ClusterDeployment) remoteclient.Builder {
 			return remoteclient.NewBuilder(c, cd, ControllerName)
 		},
+		resourceHelperBuilder: resourceHelperBuilderFunc,
 	}, nil
 }
 
-func resourceHelperBuilderFunc(restConfig *rest.Config, fakeCluster bool, logger log.FieldLogger) (resource.Helper, error) {
-	if fakeCluster {
+func resourceHelperBuilderFunc(
+	cd *hivev1.ClusterDeployment,
+	remoteClusterAPIClientBuilderFunc func(cd *hivev1.ClusterDeployment) remoteclient.Builder,
```

Comment:
It seems odd that we can't just access this via `r` from within `r.resourceHelperBuilder`... but I get it :(